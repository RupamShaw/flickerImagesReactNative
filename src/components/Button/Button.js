import React from 'react'
import {StyleSheet, View} from 'react-native'

import Touchable from 'react-native-platform-touchable'

import {Text} from '../Text/Text'
import {Image} from '../Image/Image'

export function Button(props) {
  const textColor = props.color ? props.color : 'black'
  return (
    <Touchable
      onPress={props.onPress}
      style={[
        style.btn,
        props.primary && {
          backgroundColor: props.backgroundColor || 'white',
        },
        !props.square && {
          borderRadius: 2,
        },
        !props.primary && {
          // border width set by button list
          borderWidth: props.borderWidth || 0, // no border by default
        },
      ]}
      disabled={props.disabled}
      activeOpacity={props.noTouchFeedback ? 1 : 0.2}
    >
      {props.children || (
        <View
          style={{
            flex: 1,
            opacity: props.disabled ? 0.5 : 1,
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'center',
          }}
        >
          {props.icon && (
            <View
              style={{
                paddingRight: 6,
              }}
            >
              <Image src={props.src} style={{width: 30, height: 30}} />
            </View>
          )}
          <View style={{flexShrink: 1}}>
            <Text
              // size="large"
              color={textColor}
              center
              allowFontScaling={false}
            >
              {props.title || ''}
            </Text>
          </View>
        </View>
      )}
    </Touchable>
  )
}

const style = StyleSheet.create({
  btn: {
    padding: 12,
    flexDirection: 'row',
  },
})
