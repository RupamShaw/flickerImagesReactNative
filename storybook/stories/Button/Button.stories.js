// eslint-disable-next-line import/no-extraneous-dependencies
import {action} from '@storybook/addon-actions'
// eslint-disable-next-line import/no-extraneous-dependencies
import {text} from '@storybook/addon-knobs'
// eslint-disable-next-line import/no-extraneous-dependencies
import {storiesOf} from '@storybook/react-native'
import React from 'react'
import {Text} from 'react-native'
import Button from '.'
// import CenterView from '../CenterView'

storiesOf('Button', module)
  // .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('with text', () => (
    <Button onPress={action('clicked-text')}>
      <Text>{text('Button text', 'Hello Button')}</Text>
    </Button>
  ))
  .add('with some emoji', () => (
    <Button onPress={action('clicked-emoji')}>
      <Text>😀 😎 👍 💯</Text>
    </Button>
  ))
