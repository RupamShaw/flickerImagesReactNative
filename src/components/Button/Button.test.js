import React from 'react'
import renderer from 'react-test-renderer'
import {Button} from './Button'

const testProps = {}

// eslint-disable-next-line jest/no-disabled-tests
it.skip('renders correctly snapshots', () => {
  const component = renderer.create(<Button {...testProps} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
