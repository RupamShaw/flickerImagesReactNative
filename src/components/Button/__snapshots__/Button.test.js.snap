// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly snapshots 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 2,
      "borderWidth": 0,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexShrink": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "center",
            "textDecorationLine": "none",
          }
        }
      />
    </View>
  </View>
</View>
`;
